runtime/tests/interpreter/resources_test.go

```diff
@@ -2545,3 +2545,46 @@ func TestInterpretReferenceUseAfterTransferAndDestruction(t *testing.T) {
 		assert.Equal(t, 24, invalidatedResourceErr.StartPosition().Line)
 	})
 }
+
+func TestInterpretResourceDestroyedInPreCondition(t *testing.T) {
```

Comment:
The current version of this interpreter test is always a static error, so move/copy this test case to the checker tests.

Then, adjust this interpreter test case to test what was tested before: how the interpreter handles a resource being destroyed in a pre-condition. I think this requires annotating `destroyResource` with `pure`.

Also, maybe add some whitespace between the declarations.

Reply:
Destroy is an impure operation, so destroyResource being annotated as pure would still be a static error. I am not sure how to recover the original behavior of this test to be honest, as it shouldn’t be possible anymore to destroy resources in conditions.